examples/gno.land/p/demo/avlhelpers/avlhelpers.gno

```diff
@@ -0,0 +1,32 @@
+package avlhelpers
+
+import (
+	"gno.land/p/demo/avl"
+)
+
+// Get a list of keys starting from the given prefix. Limit the
+// number of results to maxResults.
+func ListKeysByPrefix(tree avl.Tree, prefix string, maxResults int) []string {
+	end := ""
+	n := len(prefix)
+	// To make the end of the search, increment the final character ASCII by one.
+	for n > 0 {
+		if ascii := int(prefix[n-1]); ascii < 0xff {
+			end = prefix[0:n-1] + string(ascii+1)
+			break
+		}
```

Comment:
What happens if we want to search for a non-ascii prefix?

Reply:
Do you mean a mutli-byte rune (Chinese) as discussed here? https://go.dev/blog/strings

If that's what you mean, then you're right: This code won't work. What should I do about it?

Reply:
I'm guessing we'd use go's `utf8` package, something like [this](https://pkg.go.dev/unicode/utf8#DecodeLastRuneInString), but we'd need to add it into the std libs. What do you think @gfanton ?

Reply:
Hi @zivkovicmilos . I don't think we're going to be able to support searching by multi-byte runes. This is just meant to be a utility for keys in an AVL tree. It started from wanting to sort by r/demo/user usernames which have a [restricted alphabet](https://github.com/gnolang/gno/blob/master/examples/gno.land/r/demo/users/users.gno#L251). Other typical keys are bech32 addresses, or just a stringified integer.

How to more forward? A couple options.
* Document that `ListKeysByPrefix` treats the key as a byte sequence (not Unicode runes). To emphasize, maybe rename to `ListByteStringKeysByPrefix`.
* It's @moul who suggested making this a general utility. We only need it for r/demo/users for `ListUsersByPrefix`. Maybe move it back to there where it's clear that the strings are not multi-byte characters.

Reply:
I like this option:
> Document that ListKeysByPrefix treats the key as a byte sequence (not Unicode runes). To emphasize, maybe rename to ListByteStringKeysByPrefix.

Reply:
OK. I renamed to ListByteStringKeysByPrefix and added the comment "The prefix and keys are treated as byte strings, ignoring possible multi-byte Unicode runes."